EspressoSystems/nitro-testnode
# Fix the staker and contract deployment

This will be a draft PR until https://github.com/EspressoSystems/nitro-contracts/pull/10 is merged.

## This PR:
- Sync upstream
- Deploy the espresso nitro contracts instead of the original ones
- Update the espresso-dev-node

## How to test:
Run this command
```cmd
./test-node.bash --init --no-simple --espresso --latest-espresso-image
```
to see if the our contracts has been deployed and the staker is up or not.

## Remaining issues:

Currently the staker has these error logs:
> nitro-testnode-staker-unsafe-1  | INFO [07-19|05:36:34.663] rpc response                             method=eth_estimateGas logId=1102 err="err: max priority fee per gas higher than max fee per gas: address 0x6A568afe0f82d34759347bb36F14A6bB171d2CBe, maxPriorityFeePerGas: 1500000000, maxFeePerGas: 1000000000 (supplied gas 8767444)" result="\"0x0\"" attempt=0 args="[{\"from\":\"0x6a568afe0f82d34759347bb36f14a6bb171d2cbe\",\"input\":\"0x097da1f8000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000800000000000000000000000000bbc04956031f64be7ad7b5489ce2433e0238a2a0000000000000000000000000000000000000000000000000de0b6b3a764000000000000000000000000000000000000000000000000000000000000000001e44023cc4f00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000127b153d4ed7f45ce3f0c716ca5ea7c24483dc86f7eeb9af31fe5e60a9fab35a800000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000f5000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000fff3a2569dc98afc4bade748c837b9ea8478e99b0814dc4314689007fb48b30e92000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000\",\"maxFeePerGas\":\"0x3b9aca00\",\"maxPriorityFeePerGas\":\"0x59682f00\",\"to\":\"0x139a0b6b1dd1e7f912361b32a09cad89e82f29db\",\"value\":\"0xde0b6b3a7640000\"}]" errorData=null
nitro-testnode-staker-unsafe-1  | ERROR[07-19|05:36:34.663] error acting as staker                   err="getting gas for tx data: estimating gas: err: max priority fee per gas higher than max fee per gas: address 0x6A568afe0f82d34759347bb36F14A6bB171d2CBe, maxPriorityFeePerGas: 1500000000, maxFeePerGas: 1000000000 (supplied gas 8767444)"

It looks like the transaction failed because of the gas fee related configuration. Not sure which part of the code set the wrong settings. Will check it again after publish the latest image
